model/index.js

```diff
@@ -1,84 +1,183 @@
-const { Color, Size, Quantity, Shoecolor, Shoesize } = require('../db/index.js');
 const { Op } = require('sequelize');
+const {
+  Color, Size, Quantity, Shoecolor, Shoesize, Shoe,
+} = require('../db/index.js');
 
-
-let colors = (id) => {
-  return new Promise((resolve, reject) => {
-    Shoecolor.findAll({
-      where: {
-        shoe_id: id
-      }
-    })
-    .then (shoeColors => {
-      return shoeColors.map(x => x.dataValues.color_id);
-    })
-    .then(colorIDs => {
-      Color.findAll({
+const get = {
+  colors: (id) => {
+    return new Promise((resolve, reject) => {
+      Shoecolor.findAll({
         where: {
-          id: {
-            [Op.or]: colorIDs
-          }
-        }
+          shoe_id: id,
+        },
       })
-      .then(results => {
-        resolve(results.map(x => x.dataValues));
+        .then((shoeColors) => {
+          return shoeColors.map((x) => x.dataValues.color_id);
+        })
+        .then((colorIDs) => {
+          Color.findAll({
+            where: {
+              id: {
+                [Op.or]: colorIDs,
```

Comment:
I am not sure what [Op.or] does exactly

Reply:
Honestly, me either.  It's something I inherited from Jose.  It works, so I've left it.